Build failure: bear

### Steps To Reproduce

Steps to reproduce the behavior:
1. build bear

### Build log

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/blwxxfk6dk91xms2pxgchpscv54izg0y-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/aym0v0a1c7wix8fywy2bg0likz5q100s-fix-functional-tests.patch
patching file test/lit.cfg
patching script interpreter paths in test/bin
test/bin/assert_compilation: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/0nfkrvfdwxrcg0d472qqxm5cq8qr1mq4-python3-3.12.6/bin/python"
test/bin/assert_intercepted: interpreter directive changed from "#!/usr/bin/env python" to "/nix/store/0nfkrvfdwxrcg0d472qqxm5cq8qr1mq4-python3-3.12.6/bin/python"
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/share/doc/Bear -DCMAKE_INSTALL_INFODIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/include -DCMAKE_INSTALL_SBINDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/slgg1bjjvp6sswggc38yd45vlijirgai-gcc-wrapper-14.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/slgg1bjjvp6sswggc38yd45vlijirgai-gcc-wrapper-14.2.0/bin/ranlib -DCMAKE_AR=/nix/store/slgg1bjjvp6sswggc38yd45vlijirgai-gcc-wrapper-14.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/i4xipjhd36vc7kam7n5yk31zwqan5ry4-bear-3.1.3 -DCMAKE_INSTALL_BINDIR=bin -DCMAKE_INSTALL_LIBDIR=lib -DENABLE_UNIT_TESTS:BOOL=FALSE -DENABLE_FUNC_TESTS:BOOL=FALSE
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
[...]
-- Configuring done (0.6s)
-- Generating done (0.0s)
[33mCMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_NAME_DIR
    CMAKE_POLICY_DEFAULT_CMP0025

[0m
-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j32
[1/9] Creating directories for 'BearSource'
[2/9] No download step for 'BearSource'
[3/9] No update step for 'BearSource'
[4/9] No patch step for 'BearSource'
[5/9] Performing configure step for 'BearSource'
loading initial cache file /build/source/build/subprojects/tmp/BearSource/BearSource-cache-Release.cmake
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
[...]
-- Configuring done (2.1s)
-- Generating done (0.0s)
-- Build files have been written to: /build/source/build/subprojects/Build/BearSource
[6/9] Performing build step for 'BearSource'
FAILED: subprojects/Stamp/BearSource/BearSource-build /build/source/build/subprojects/Stamp/BearSource/BearSource-build 
cd /build/source/build/subprojects/Build/BearSource && /nix/store/98zigx84f1x53vyn1xc0y8g4ml5dzfsf-cmake-3.29.6/bin/cmake --build . && /nix/store/98zigx84f1x53vyn1xc0y8g4ml5dzfsf-cmake-3.29.6/bin/cmake -E touch /build/source/build/subprojects/Stamp/BearSource/BearSource-build
[1/59] Building CXX object intercept/CMakeFiles/exec.dir/source/report/libexec/std.cc.o
[...]
[33/59] Building CXX object intercept/CMakeFiles/events_db_a.dir/source/collect/db/EventsDatabaseWriter.cc.o
FAILED: intercept/CMakeFiles/events_db_a.dir/source/collect/db/EventsDatabaseWriter.cc.o 
/nix/store/slgg1bjjvp6sswggc38yd45vlijirgai-gcc-wrapper-14.2.0/bin/g++ -DFMT_SHARED -DSPDLOG_COMPILED_LIB -DSPDLOG_FMT_EXTERNAL -DSPDLOG_SHARED_LIB -I/build/source/build/subprojects/Build/BearSource -I/build/source/source/intercept/source -I/build/source/source/libresult/include -I/build/source/build/subprojects/Build/BearSource/intercept/proto -I/build/source/source/libsys/include -fno-exceptions -DSPDLOG_NO_EXCEPTIONS -DGOOGLE_PROTOBUF_NO_RTTI -Wall -Wextra -O3 -DNDEBUG -std=c++17 -DPROTOBUF_USE_DLLS -Wnon-virtual-dtor -DNOMINMAX -MD -MT intercept/CMakeFiles/events_db_a.dir/source/collect/db/EventsDatabaseWriter.cc.o -MF intercept/CMakeFiles/events_db_a.dir/source/collect/db/EventsDatabaseWriter.cc.o.d -o intercept/CMakeFiles/events_db_a.dir/source/collect/db/EventsDatabaseWriter.cc.o -c /build/source/source/intercept/source/collect/db/EventsDatabaseWriter.cc
/build/source/source/intercept/source/collect/db/EventsDatabaseWriter.cc: In function 'google::protobuf::util::JsonPrintOptions {anonymous}::create_print_options()':
/build/source/source/intercept/source/collect/db/EventsDatabaseWriter.cc:39:23: error: 'using google::protobuf::util::JsonPrintOptions = struct google::protobuf::json::PrintOptions' {aka 'struct google::protobuf::json::PrintOptions'} has no member named 'always_print_primitive_fields'
   39 |         print_options.always_print_primitive_fields = true;
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
ninja: build stopped: subcommand failed.
ninja: build stopped: subcommand failed.

```

### Additional context

Seems to have been broken by #338885. 

It broke between 476bcb4bf099a04ad41e3c0590445ef0e1dea5d4...73057677e8557925e999ac54196423fa34418c24, the last nixos-unstable-small push.

Explicitly depending on `protobuf_25` instead of `protobuf` fixes the problem.

### Notify maintainers

@DieracDelta 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix shell nixpkgs#nix-info -c nix-info -m
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.1, NixOS, 24.11 (Vicuna), 24.11.20241012.162ee41`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"nixos"`
 - channels(john): `""`
 - nixpkgs: `/nix/store/sdzpqjwx7pdx6lsq6llyfqqf7hspp83c-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/tools/build-managers/bear/default.nix
+++ b/pkgs/development/tools/build-managers/bear/default.nix
@@ -1,21 +1,23 @@
-{ lib, stdenv
-, fetchFromGitHub
-, cmake
-, ninja
-, pkg-config
-, grpc
-, protobuf
-, openssl
-, nlohmann_json
-, gtest
-, spdlog
-, c-ares
-, zlib
-, sqlite
-, re2
-, lit
-, python3
-, coreutils
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  pkg-config,
+  grpc,
+  protobuf,
+  openssl,
+  nlohmann_json,
+  gtest,
+  spdlog,
+  c-ares,
+  zlib,
+  sqlite,
+  re2,
+  lit,
+  python3,
+  coreutils,
 }:
 
 stdenv.mkDerivation rec {

--- a/pkgs/development/tools/build-managers/bear/default.nix
+++ b/pkgs/development/tools/build-managers/bear/default.nix
@@ -6,7 +6,7 @@
   ninja,
   pkg-config,
   grpc,
-  protobuf,
+  protobuf_25,
   openssl,
   nlohmann_json,
   gtest,
@@ -43,7 +43,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     grpc
-    protobuf
+    protobuf_25
     openssl
     nlohmann_json
     gtest

